simple/kubernetes-apim-analytics/dashboard/wso2am-pattern-1-analytics-dashboard-deployment.yaml

```diff
@@ -0,0 +1,77 @@
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: wso2am-pattern-1-analytics-dashboard-deployment
+  namespace: wso2
+spec:
+  replicas: 1
+  minReadySeconds: 75
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      deployment: wso2am-pattern-1-analytics-dashboard
+      product: api-manager
+  template:
+    metadata:
+      labels:
+        deployment: wso2am-pattern-1-analytics-dashboard
+        product: api-manager
+    spec:
+      initContainers:
+        - name: init-apim-analytics-db
+          image: busybox:1.31
+          command: ['sh', '-c', 'echo -e "Checking for the availability of MySQL Server deployment"; while ! nc -z wso2apim-rdbms-service-mysql 3306; do sleep 1; printf "-"; done; echo -e "  >> MySQL Server has started";']
+      containers:
+        - name: wso2am-pattern-1-analytics-dashboard
+          image: "$image.pull.@.wso2"/wso2am-analytics-dashboard:3.0.0
+          livenessProbe:
+            exec:
+              command:
+                - /bin/sh
+                - -c
+                - nc -z localhost 30643
+            initialDelaySeconds: 100
```

Comment:
@NishikaDeSilva IMO the following default period values and time intervals would suffice for Analytics profiles.

```
        # Minimum number of seconds for which a newly created pod should be ready without any of its containers crashing, for it to be considered available.
        minReadySeconds: 30
        strategy:
          rollingUpdate:
            # The maximum number of pods that can be scheduled above the desired number of pods.
            maxSurge: 1
            # The maximum number of pods that can be unavailable during the update.
            maxUnavailable: 0
        # Indicates whether the container is running.
        livenessProbe:
          # Number of seconds after the container has started before liveness probes are initiated.
          initialDelaySeconds: 20
          # How often (in seconds) to perform the probe.
          periodSeconds: 10
        # Indicates whether the container is ready to service requests.
        readinessProbe:
          # Number of seconds after the container has started before readiness probes are initiated.
          initialDelaySeconds: 20
          # How often (in seconds) to perform the probe.
          periodSeconds: 10
```

Reply:
@chirangaalwis updated